MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,254 @@
+            /// </summary>
+            public Random()
+                : this(GenerateSeed())
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the random number generator using the specified seed
+            /// </summary>
+            /// <param name="seed">Seed to be used when computing new random numbers</param>
+            [ClsCompliant(false)]
+            public Random(ulong seed)
+            {
+                this.Seed = seed;
+            }
+
+            #endregion
+
+            #region Public methods
+
+            #region Int8 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int8 (byte) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int8 (byte)</returns>
+            public byte Next(byte max)
+            {
+                return (byte)Remap(UInt64ToInt8(XORShift64Star()), 0, byte.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int8 (byte) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int8 (byte)</returns>
+            public byte Next(byte min, byte max)
+            {
+                return (byte)Remap(UInt64ToInt16(XORShift64Star()), 0, byte.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #region Int16 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int16 (short) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short Next(short max)
+            {
+                return (short)Remap(UInt64ToInt16(XORShift64Star()), short.MinValue, short.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int16 (short) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short Next(short min, short max)
+            {
+                return (short)Remap(UInt64ToInt16(XORShift64Star()), short.MinValue, short.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #region Int32 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int).
```

Comment:
Yeah, they are allowed in the ones with minimum specified already. Only the ones with no min. specified default to zero, mimicking the System.Random behavior.